File: Android/_Viper Fragment_/Code/user_interface/VIPERRouter.java
//
//  __MODULENAME__Router.java
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.user_interface;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import NAMESPACE.modules._VIPER_.contracts.VIPERContracts;

public class VIPERRouter implements VIPERContracts.Router {

    private FragmentActivity activity;

    VIPERRouter(Fragment fragment) {
        this.activity = fragment.requireActivity();
    }

    @Override
    public void unload() {
        this.activity = null;
    }
}

File: Kotlin/_VIPER Fragment_/Code/user_interface/VIPERRouter.kt
//
//  __MODULENAME__Router.java
//  __APPNAME__
//
//  Created by __USERNAME__ on __DATETIME__.
//  Copyright © __YEAR__ __ORGANIZATIONNAME__. All rights reserved.
//

package NAMESPACE.modules._VIPER_.user_interface

import androidx.fragment.app.Fragment
import NAMESPACE.modules._VIPER_.contracts.VIPERContracts.Router
import androidx.fragment.app.FragmentActivity

class VIPERRouter internal constructor(fragment: Fragment) : Router {
    private var activity: FragmentActivity?
    
    override fun unload() {
        activity = null
    }

    init {
        activity = fragment.requireActivity()
    }
}